### PR TITLE
Release notes for 2.17.0

### PR DESCRIPTION
**Pull request by desertaxle**
[RENDERED](https://github.com/PrefectHQ/prefect/blob/3c86d55a1f4317fb577deb81723ecab4c724058c/RELEASE-NOTES.md)

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,89 @@
 # Prefect Release Notes
 
+## Release 2.17.0
+
+### Manage Prefect variables via the Python SDK
+
+Prefect variables are useful for storing and reusing data and configuration between and across workflows, but previously you could only create and update variables via the Prefect UI. With this release, you can now get and set Prefect variables directly in your Python code with the new `Variable.set` and `Variable.get` methods!
```
Comment (serinamarie) [inline RELEASE-NOTES.md:7]:
```suggestion
Prefect variables are useful for storing and reusing data and configuration between and across workflows; and previously you could only create and update variables via the Prefect UI. With this release, you can now get and set Prefect variables directly in your Python code with the new `Variable.set` and `Variable.get` methods!
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,89 @@
...
+
+For an example of reading and writing variable values in Python see the following example:
+
+```python
+from prefect.variables import Variable
+
+# set a variable
+variable = Variable.set(name="the_answer", value="42")
+
+# get a variables
```
Comment (serinamarie) [inline RELEASE-NOTES.md:17]:
```suggestion
# get a variable
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,89 @@
...
+answer = Variable.get('the_answer')
+print(answer.value)
+# 42
+
+# getting with a default value
+answer = Variable.get('not_the_answer', default='42')
+print(answer.value)
+# 42
+
+# updating a variable
```
Comment (serinamarie) [inline RELEASE-NOTES.md:27]:
```suggestion
# update a variable
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,89 @@
...
+answer = Variable.set(name="the_answer", value="43", overwrite=True)
+print(answer.value)
+#43
+```
+
+See the PR for implementation details: https://github.com/PrefectHQ/prefect/pull/12596
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:34]: Would love if we could point to the docs :) https://docs.prefect.io/latest/guides/variables/#accessing-variables

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,89 @@
...
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:35]:
```suggestion
```

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,5 +1,89 @@
...
+- Remove duplicate CLI output when reauthorizing with `prefect cloud login` — https://github.com/PrefectHQ/prefect/pull/12664
+- Add `blob_storage` extra as requirement for Azure recipes — https://github.com/PrefectHQ/prefect/pull/12333
+- Exclude Typer 0.12.2 from solver — https://github.com/PrefectHQ/prefect/pull/12618
+- Correct `schedules`/`is_schedule_active` deprecation windows — https://github.com/PrefectHQ/prefect/pull/12616
+
+### Experimental / In-Flight Features
+
+#### Pydantic V2 Compatibility
+
+- Add `pydantic` V2 compatible field validator  — https://github.com/PrefectHQ/prefect/pull/12576
```
Comment (serinamarie) [inline RELEASE-NOTES.md:54]:
```suggestion
- Add `pydantic` V2 compatible `field_validator`  — https://github.com/PrefectHQ/prefect/pull/12576
```

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,5 +1,89 @@
...
+- Add `pydantic` V2 `model_validator` — https://github.com/PrefectHQ/prefect/pull/12635
+- Expose `field_validator` in `pydantic` compatibility layer — https://github.com/PrefectHQ/prefect/pull/12608
+- Add `ConfigDict` to `pydantic` compatibility layer — https://github.com/PrefectHQ/prefect/pull/12629
+- Add `model_fields_set` to `pydantic` compatibility layer — https://github.com/PrefectHQ/prefect/pull/12654
+- Map `copy_on_model_validation` to `revalidate_instances` in `pydantic` compatibility layer — https://github.com/PrefectHQ/prefect/pull/12644
+
+#### Events and Automations
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:62]:
```suggestion
```

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,5 +1,89 @@
...
+- Add orchestration rule to transition flow runs without active infrastructure directly to cancelled — https://github.com/PrefectHQ/prefect/pull/12582
+- Remove duplicate CLI output when reauthorizing with `prefect cloud login` — https://github.com/PrefectHQ/prefect/pull/12664
+- Add `blob_storage` extra as requirement for Azure recipes — https://github.com/PrefectHQ/prefect/pull/12333
+- Exclude Typer 0.12.2 from solver — https://github.com/PrefectHQ/prefect/pull/12618
+- Correct `schedules`/`is_schedule_active` deprecation windows — https://github.com/PrefectHQ/prefect/pull/12616
+
+### Experimental / In-Flight Features
+
+#### Pydantic V2 Compatibility
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:53]:
```suggestion
```

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,5 +1,89 @@
...
+### Enhancements
+- Allow flows inside tasks — https://github.com/PrefectHQ/prefect/pull/12559
+- Add `User-Agent` header containing the running Prefect version — https://github.com/PrefectHQ/prefect/pull/12601
+- Add `Variable.get` and `Variable.set` for retrieving and updating Prefect variable values — https://github.com/PrefectHQ/prefect/pull/12596
+- Adds deployment version to the flow run object — https://github.com/PrefectHQ/prefect/pull/12591
+- Allow flows in standalone tasks — https://github.com/PrefectHQ/prefect/pull/12607
+
+### Fixes
+- Add orchestration rule to transition flow runs without active infrastructure directly to cancelled — https://github.com/PrefectHQ/prefect/pull/12582
```
Comment (serinamarie) [inline RELEASE-NOTES.md:44]:
```suggestion
- Transition flow runs without active infrastructure directly to cancelled — https://github.com/PrefectHQ/prefect/pull/12582
```

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,5 +1,89 @@
...
+- Remove duplicate CLI output when reauthorizing with `prefect cloud login` — https://github.com/PrefectHQ/prefect/pull/12664
+- Add `blob_storage` extra as requirement for Azure recipes — https://github.com/PrefectHQ/prefect/pull/12333
```
Comment (serinamarie) [inline RELEASE-NOTES.md:46]:
```suggestion
- Add `blob_storage` extra as requirement for Azure `prefect.yaml` recipes — https://github.com/PrefectHQ/prefect/pull/12333
```

File: RELEASE-NOTES.md [hunk 11]
```diff
@@ -1,5 +1,89 @@
...
+#43
+```
+
+See the PR for implementation details: https://github.com/PrefectHQ/prefect/pull/12596
+
+
+### Enhancements
+- Allow flows inside tasks — https://github.com/PrefectHQ/prefect/pull/12559
+- Add `User-Agent` header containing the running Prefect version — https://github.com/PrefectHQ/prefect/pull/12601
+- Add `Variable.get` and `Variable.set` for retrieving and updating Prefect variable values — https://github.com/PrefectHQ/prefect/pull/12596
```
Comment (serinamarie) [inline RELEASE-NOTES.md:39]: We can remove this here since it's in the highlight

File: RELEASE-NOTES.md [hunk 12]
```diff
@@ -1,5 +1,89 @@
...
+- Adds deployment version to the flow run object — https://github.com/PrefectHQ/prefect/pull/12591
```
Comment (serinamarie) [inline RELEASE-NOTES.md:40]:
```suggestion
- Add deployment version to the flow run object — https://github.com/PrefectHQ/prefect/pull/12591
```

File: RELEASE-NOTES.md [hunk 13]
```diff
@@ -1,5 +1,89 @@
...
+answer = Variable.set(name="the_answer", value="43", overwrite=True)
+print(answer.value)
+#43
+```
+
+See the PR for implementation details: https://github.com/PrefectHQ/prefect/pull/12596
+
+
+### Enhancements
+- Allow flows inside tasks — https://github.com/PrefectHQ/prefect/pull/12559
```
Comment (serinamarie) [inline RELEASE-NOTES.md:37]: Should this one and #12607 be combined under one heading? I'm not totally sure on the diff

Comment (desertaxle) [reply RELEASE-NOTES.md]: That makes sense. I combined them.

**Pull request merged**